src/utils.cpp

```diff
@@ -1176,18 +1176,21 @@ std::string utils::get_basename(const std::string& url)
 	char* base = NULL;
```

Comment:
Since this variable is no longer used outside of nested `if`s, please remove it and declare `base` on line 1182.

Also, please change that `NULL` to `nullptr`; we're using C++11 now ;)